graphql_compiler/query_pagination/pagination_planning.py

```diff
@@ -1,53 +1,156 @@
 # Copyright 2019-present Kensho Technologies, LLC.
-from collections import namedtuple
+from typing import List, NamedTuple, Tuple
+
+from graphql import DocumentNode
 
 from ..ast_manipulation import get_only_query_definition, get_only_selection_from_ast
-from ..cost_estimation.helpers import is_int_field_type, is_uuid4_type
+from ..cost_estimation.helpers import is_uuid4_type
+from ..cost_estimation.int_value_conversion import field_supports_range_reasoning
 from ..exceptions import GraphQLError
+from ..schema.schema_info import QueryPlanningSchemaInfo
+
+
+class PaginationWarning(object):
+    def __init__(self, message):
+        """Initialize a PaginationWarning with a custom message."""
+        self.message = message
+
+
+class PaginationFieldNotSpecified(PaginationWarning):
+    def __init__(self, vertex_name):
+        """Initialize a PaginationFieldNotSpecified PaginationWarning."""
+        super(PaginationFieldNotSpecified, self).__init__(
+            "Pagination field not specified for vertex {}".format(vertex_name)
```

Comment:
Consider adding the suggested resolution steps here, similarly to how you've done below.